src/gc.c

```diff
@@ -878,7 +878,9 @@ void hl_free( hl_alloc *a ) {
 		size = (int)(b->p + b->size - ((unsigned char*)b));
 		prev = (int_val)b;
 		free(b);
-		b = n;
+		if (n) {
+			b = n;
+		}

```

Comment:
This change seems wrong, and might be why CI is failing. This `while` is walking down a linked list, so it must terminate when `n == b->next == NULL`. If you avoid setting `b` for the final iteration, then `b` will always be the last node, so the loop will run forever; it is in fact stopped first because `free` rightly complains that the same `b` is `free`d again.

Reply:
Hmmmm, makes sense, I'll see if I find a way to fix the warnings without breaking the CI.

Reply:
I cannot think of how to fix this warning or the one with the i64 structure in src/std/bytes.c. So for now I am going to leave my fixes down to this point, as long as I haven't broken anything else, which I should the fix.